File: Leetcode Challenge/April/C++/Trim a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* trimBST(TreeNode* root, int low, int high) {
        //Base case
       /**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* trimBST(TreeNode* root, int low, int high) {
        //Base case
        if(root == NULL) {
            return NULL;
        }
        
        if(root->val >= low && root->val <= high) {
            root->left = trimBST(root->left, low, high);
            root->right = trimBST(root->right, low, high);
            return root;
        }
        
        else if(root->val > high) {
            return trimBST(root->left, low, high);
        }
        else if(root->val < low) {
            return trimBST(root->right, low, high);
        }
        
        return root;
    }
};
    }
};

File: Leetcode Challenge/April/Python/Trim a BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    
    def trimBST(self, root: TreeNode, L: int, R: int) -> TreeNode:
        
        if root==None:
            return None
        
        if root.val==L:
            root.left=None
            root.right=self.trimBST(root.right, L, R)
            return root
        
        
        if root.val==R:
            root.right=None
            root.left=self.trimBST(root.left, L, R)
            return root
        
        
        if root.val>R and root.val>L:
            return self.trimBST(root.left, L, R)
        
        if root.val<L and root.val<R:
            return self.trimBST(root.right, L, R)
       
    
        root.left=self.trimBST(root.left, L, R)
        root.right=self.trimBST(root.right, L, R)
        
        return root
